src/plugins/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -114,17 +114,22 @@ function DiscoverDocumentsComponent({
   const sampleSize = useMemo(() => uiSettings.get(SAMPLE_SIZE_SETTING), [uiSettings]);
 
   const documentState = useDataState(documents$);
-  const isDataLoading = documentState.fetchStatus === FetchStatus.LOADING;
+  const isDataLoading =
+    documentState.fetchStatus === FetchStatus.LOADING ||
+    documentState.fetchStatus === FetchStatus.PARTIAL;
```

Comment:
PARTIAL state is not also considered as loading state in the UI (for text based queries)